calicoctl/commands/printer.go

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) 2016 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"fmt"
+	"io/ioutil"
+	"reflect"
+	"strings"
+
+	"bytes"
+	"encoding/json"
+	"os"
+	"text/tabwriter"
+	"text/template"
+
+	"github.com/ghodss/yaml"
+	"github.com/golang/glog"
+	"github.com/tigera/libcalico-go/calicoctl/resourcemgr"
+	"github.com/tigera/libcalico-go/lib/api/unversioned"
+)
+
+type resourcePrinter interface {
+	print(resources []unversioned.Resource) error
+}
+
+// JSON output formatter
+type resourcePrinterJSON struct{}
+
+func (r resourcePrinterJSON) print(resources []unversioned.Resource) error {
+	// For simplicity convert the slice of resources (which may contain resource lists)
+	// to expand any lists - so we just work with a slice of real resource types.
+	resources = convertToSliceOfResources(resources)
```

Comment:
I'll try and clarify the text a little - I have an issue to get rid of this code at some point.